scripts/common/actor.js

```diff
@@ -224,6 +224,74 @@ export class AgeOfSigmarActor extends Actor {
         this.power.isUndercharge =             this.power.consumed > this.power.capacity;
     }
 
+    /**
+     * applies Damage to the actor
+     * @param {int} damages 
+     */
+    async applyDamage(damage) {
+        let remaining = this.combat.health.toughness.value - damage;
+         // Update the Actor
+         const updates = {
+            "data.combat.health.toughness.value": remaining >= 0 ? remaining : 0
+        };
+
+        // Delegate damage application to a hook
+        const allowed = Hooks.call("modifyTokenAttribute", {
+            attribute: "combat.health.toughness.value",
+            value: this.combat.health.toughness.value,
+            isDelta: false,
+            isBar: true
+        }, updates);
+
+        let ret = allowed !== false ? this.update(updates) : this;
+
+        // WHY DOES String.Format NOT EXIST IN JAVASCRIPT
+        let note = game.i18n.localize("NOTIFICATION.APPLY_DAMAGE").replace("{0}", damage).replace("{1}", this.data.token.name);
```

Comment:
You can use foundry's localization formatter:

`game.i18n.format("NOTIFICATION.APPLY_DAMAGE", {damage : damage, name : this.data.token.name})`

` "NOTIFICATION.APPLY_DAMAGE": "{damage} Damage applied to {name}"`

Reply:
neat